### PR TITLE
chore(linters): Bump golangci-lint to v1.56.0

### PR DESCRIPTION
**Pull request by zak-pawel**
- `golangci-lint` bumped to `v1.56.0`
- `sloppyTestFuncName` checker has been removed from `go-critic`: https://github.com/go-critic/go-critic/pull/1376
- `disable-all` set for `testifylint` before enabling specific checkers (without it, not needed checkers enabled by default would start working)
- new feature: `show-stats` enabled
- few `//nolint:testifylint // https://github.com/Antonboom/testifylint/issues/6` removed (issue was fixed and new version of `testifylint` was included in `golangci-lint`)
- new findings have been addressed:
```
config/types_test.go:30:2                              testifylint  expected-actual: need to reverse actual and expected values
config/types_test.go:31:2                              testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/burrow/burrow_test.go:194:2             testifylint  len: use require.Len
plugins/inputs/burrow/burrow_test.go:210:3             testifylint  len: use require.Len
plugins/inputs/burrow/burrow_test.go:229:2             testifylint  len: use require.Len
plugins/inputs/burrow/burrow_test.go:247:2             testifylint  empty: use require.Empty
plugins/inputs/burrow/burrow_test.go:265:2             testifylint  len: use require.Len
plugins/inputs/burrow/burrow_test.go:283:2             testifylint  len: use require.Len
plugins/inputs/cloudwatch/cloudwatch_test.go:403:2     testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/cloudwatch/cloudwatch_test.go:444:2     testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/ecs/types_test.go:45:3                  testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/ipmi_sensor/ipmi_sensor_test.go:33:2    testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/opcua/opcua_test.go:435:2               testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/opcua/opcua_test.go:436:2               testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/opcua/opcua_test.go:437:2               testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/opcua/opcua_test.go:438:2               testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/opcua/opcua_test.go:439:2               testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/statsd/statsd_test.go:1293:2            testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/statsd/statsd_test.go:1295:2            testifylint  expected-actual: need to reverse actual and expected values
plugins/inputs/zfs/zfs_linux_test.go:394:9             perfsprint   fmt.Sprintf can be replaced with just using the string
plugins/inputs/zipkin/codec/jsonV1/jsonV1.go:238:10    perfsprint   fmt.Sprintf can be replaced with faster strconv.FormatUint
plugins/outputs/stackdriver/stackdriver_test.go:404:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/stackdriver/stackdriver_test.go:409:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/stackdriver/stackdriver_test.go:419:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/stackdriver/stackdriver_test.go:424:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/stackdriver/stackdriver_test.go:537:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/stackdriver/stackdriver_test.go:542:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/stackdriver/stackdriver_test.go:550:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/stackdriver/stackdriver_test.go:555:2  testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/timestream/timestream_test.go:422:2    testifylint  expected-actual: need to reverse actual and expected values
plugins/outputs/timestream/timestream_test.go:447:2    testifylint  expected-actual: need to reverse actual and expected values

```

### CODE REVIEW

**Review by srebhan — APPROVED** [review]
Thanks @zak-pawel!

File: plugins/inputs/ecs/types_test.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func Test_mergeTaskStats(t *testing.T) {
 	mergeTaskStats(parsedMetadata, parsedStats)
 
 	for i := range parsedMetadata.Containers {
+		//nolint:testifylint // values are in proper order (actual, expected)
 		require.Equal(t, validStats[parsedMetadata.Containers[i].ID], &parsedMetadata.Containers[i].Stats)
```
Comment (DStrand1) [inline plugins/inputs/ecs/types_test.go:46]: The values here do seem to be in the correct order so the lint suppression is correct, but the comment is written backwards

**Review by DStrand1 — COMMENTED** [review]
Thanks @zak-pawel! Just one very minor comment which caused me a little confusion while reviewing

File: plugins/inputs/ecs/types_test.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/ecs/types_test.go]: Of course, my bad.

**Review by srebhan — APPROVED** [review]
Good catch @DStrand1!

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**